File: 브루트포스/1182.py
import sys
input = sys.stdin.readline

n,s = map(int,input().rstrip().split())
l = list(map(int,input().rstrip().split()))
cnt = 0
def dfs(idx, total):
    global cnt
    if idx ==n:
        if total==s:
            cnt +=1
        return
    dfs(idx+1,total+l[idx])
    dfs(idx+1,total)

dfs(0,0)
if s == 0:
    cnt -=1

print(cnt)

File: 브루트포스/1182.java
import java.util.*;

public class Main {
    public static int dfs(int[] a,int m, int idx, int sum) {
        if(idx == a.length){
            if(sum == m) {
                return 1;
            }
            else {
                return 0;
            }
        }
        return dfs(a,m,idx+1,sum+a[idx]) + dfs(a,m,idx+1,sum);
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int s = sc.nextInt();
        int[] a = new int[n];
        for(int i = 0;i<n;i++){
            a[i] = sc.nextInt();
        }
        int cnt = dfs(a,s,0,0);
        if(s==0) {
            cnt--;
        }
        System.out.println(cnt);
    }
}